hw/ip/i2c/doc/_index.md

```diff
@@ -1,3 +1,225 @@
+When {{< regref OVRD.SCLVAL >}} and {{< regref OVRD.SDAVAL >}} are set high, the virtual open drain configuration will leave the output resistively pulled high, and controllable by remote targets.
+In this state, with SCL or SDA asserted high, the register fields {{< regref VAL.SCL_RX >}} and {{< regref VAL.SDA_RX >}} can be used to receive inputs (including remote acknowledgments) from target devices.
+
+### Byte-Formatted Programming Mode 
+
+The state machine-controlled mode allows for higher-speed operation with less frequent software interaction.
+In this mode, the I2C pins are controlled by the I2C state machine, which in turn is controlled by a sequence of formatting indicators.
+The formatting indicators indicate:
+- The sequence of bytes which should be transmitted on the SDA and SCL pins.
+- The periods between transmitted bytes when the state-machine should stop transmission and instead read back a fixed number of bytes.
+- Which bytes should be preceded by a START symbol.
+- Which bytes should be followed by a STOP symbol
+The format indicator consists of 13-bits. That is of one single Format Byte (entered into the format FIFO through {{< regref FDATA.FBYTE >}}), and five (5) 1-bit flags (entered into the format FIFO through registers {{< regref FDATA.READ >}}, {{< regref FDATA.RCONT >}}, {{< regref FDATA.START >}}, {{< regref FDATA.STOP >}} and {{< regref FDATA.NAKOK >}})
+
+The I2C reads each format indicator from the head of FMT_FIFO, and processes them in turn.
+If none of the flags are set for the format format indicator none, the I2C FSM simply transmits the Format Byte onto the SCL and SDA pins according to the specification, waits for acknowledgement, and then proceeds to the next format indicator.
+The format flags modulate the behavior as follows.
+- READ (corresponds to {{< regref FDATA.READ >}}):
+Signifies the Format Byte ({{< regref FDATA.FBYTE >}}) should be treated as an unsigned number, R, and prompts the state machine to read R bytes from the target device.
+Bytes read from the bus, are inserted into the RX FIFO where they can be accessed by software.
+A value of 0 is treated as a read of 256B.
+To read a larger byte stream, multiple 256B reads can be chained together using the RCONT flag.
+- RCONT (corresponds to FIFO inputs {{< regref FDATA.RCONT >}}, only used with READ):
+If RCONT is set, the Format Byte represents part of a longer sequence of reads, allowing for reads to be chained indefinitely.
+The RCONT flag indicates the the final byte returned with the current read
+should be responded to which an ACK, allowing the target to continue sending data.
+(Note that the first R-1 bytes read will still be acknowledged regardless of whether RCONT is asserted or not.)
+- START (corresponds to {{< regref FDATA.START >}}, Ignored when used with READ):
+Issue a START condition before transmitting the Format Byte on the bus.
```

Comment:
ditto: sub-bullet for better readability?

Reply:
> ditto: sub-bullet for better readability?

Hi @sjgitty, 
   If you have a chance, take a look at the latest and see if that is what you had in mind.  The current version looks nice in the render.

Reply:
yep, looks good thanks